packages/react/src/prop-types/deprecateValuesWithin.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright IBM Corp. 2016, 2023
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { warning } from '../internal/warning';
+
+const didWarnAboutDeprecation = {};
+
+export default function deprecateValuesWithin(
+  propType,
+  message,
+  allowedValues
+) {
+  return function checker(props, propName, componentName, ...rest) {
+    if (props[propName] === undefined) {
+      return;
+    }
+
+    if (
+      !didWarnAboutDeprecation[componentName] ||
+      !didWarnAboutDeprecation[componentName][propName]
+    ) {
+      didWarnAboutDeprecation[componentName] = {
+        ...didWarnAboutDeprecation[componentName],
+        [propName]: true,
+      };
+
+      if (allowedValues && !allowedValues.includes(props[propName])) {
+        warning(
+          false,
+          message ||
+            `In \`${componentName}\` component value for \`${propName}\` prop has been deprecated, It will be removed in the next major release. Allowed values is/are: ${allowedValues.join(
```

Comment:
Here we can optionally call that mapper function so we can guide the user more clearly and cut down the message length. If we don't have the mapper, we can instead provide a list of all the valid options.


```suggestion
            `\`${props[propName]}\` is a deprecated value for the \`${propName}\` prop on the \`${componentName}\` component. \`${propMappingFunction ? \`Use ${propMappingFunction(props[propName])} instead\` : \`The allowed values for \`${propName}\` include: ${allowedValues.join(', ')}\`}\`. \`${props[propName]}\` will be removed in the next major release.\`
```

With the mapper:
> `top-left` is a deprecated value for the `align` prop on the `Popover` component. Use `top-start` instead. `top-left` will be removed in the next major release. 

Without the mapper:
> `top-left` is a deprecated value for the `align` prop on the `Popover` component. The allowed values for `align` include: 'top', 'top-start', '...'. `top-left` will be removed in the next major release.

Reply:
Agreed , this would look so much better and clear to users 🔥 .
I will take up this enhancement.